test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/CompilerCacheTest.cs

```diff
@@ -54,12 +72,12 @@ public void GetOrAdd_ReturnsCompilationResultFromFactory()
             var expected = new CompilationResult(typeof(TestView));
 
             // Act
-            var result = cache.GetOrAdd(ViewPath, _ => expected);
+            var result = cache.GetOrAdd(ViewPath, CreateContextFactory(expected));
 
             // Assert
             Assert.True(result.Success);
-            Assert.IsType<TestView>(result.PageFactory());
-            Assert.Same(ViewPath, result.PageFactory().Path);
+            Assert.Equal(typeof(TestView), result.CompiledType);
```

Comment:
I see it is just a not-null check. But just to put it out there, is it worth adding `Assert.True(result.Success)`?